frontend/dockerfile/instructions/parse.go

```diff
@@ -815,3 +828,40 @@ func allInstructionNames() []string {
 	}
 	return out
 }
+
+func isLowerCased(s string) bool {
```

Comment:
nit: I don't think `isLowerCased` and `isUpperCased` are needed. This is simple enough to be inline and more readable without extra wrapping.

Reply:
```
type casing int

const (
	caseAny casing = iota
	caseLower
	caseUpper
)

func checkCase(s string, exp casing) (consistent, isLower bool) {
	expLower := exp == caseLower
	for _, r := range s {
		if l := unicode.IsLower(r); exp == caseAny {
			expLower = l
			exp = caseLower
		} else if expLower != l {
			return false, false
		}
	}
	return true, expLower
}
```

could maybe also be an option but probably unnecessary